src/d3d9/d3d9_device.cpp

```diff
@@ -3705,12 +3701,10 @@ namespace dxvk {
       // that cannot get affected by GPU, therefore readonly is A-OK for NOT waiting.
       const bool noOverwrite  = Flags & D3DLOCK_NOOVERWRITE;
       const bool readOnly     = Flags & D3DLOCK_READONLY;
-      const bool gpuImmutable = (readOnly && managed) || scratch || (systemmem && !modified);
```

Comment:
The read only rule only applies to managed here.

Just get rid of the `(systemmem && !modified)` bit.

Reply:
Does it?

Either way, we also need to wait for scratch as it can also be locked and copied.

Reply:
> Because of this, these resources cannot be accessed by the Direct3D device nor set as textures or render target

They will never be accessed by the CS thread? Unless there's an edgecase I am not considering.

Reply:
> However, these resources can always be created, locked, and copied.

We do implement copying using the GPU though. (Unless Microsoft means copying the data after locking them)

Reply:
I can't think of a function that allows D3DPOOL_SCRATCH to be copied.

Neither UpdateTexture nor UpdateSurface allow it.

Reply:
I guess that's fine then. Got any source on the readonly only for managed thing?

Reply:
The same reason for D3DPOOL_SCRATCH above

Reply:
?

SystemMem texture can be accessed by the D3D9 device and used for copying. That's the whole issue here. Why wouldn't games be able to tell the driver that they're just gonna read from it?

Reply:
We're talking about managed here?

Reply:
Discussion resolved on Discord